internal/core/data/valuedescriptor.go

```diff
@@ -200,7 +201,8 @@ func decodeValueDescriptor(reader io.ReadCloser) (vd contract.ValueDescriptor, e
 		return contract.ValueDescriptor{}, err
 	}
 	if !match {
```

Comment:
It seems like this check and the handling for it should be in the validateFormatString() function itself. In which case, the function would just return an error of type `ErrValueDescriptorInvalid`. That would get logged and returned.

Reply:
Thanks, that makes a lot of sense.  I think I've addressed this now.